scripts/submit-analytics.js

```diff
@@ -0,0 +1,208 @@
+        Binding: "javascript",
+        Version: context.version,
+        Language: "javascript",
+        Framework: framework,
+        "Framework Version": frameworkVersion,
+        "JavaScript Engine": jsEngine,
+        "Host OS Type": os.platform(),
+        "Host OS Version": os.release(),
+        "Node.js version": process.version,
+      },
+    },
+  };
+
+  return payload;
+}
+
+function isAnalyticsDisabled() {
+  let isDisabled = false;
+
+  // NODE_ENV is commonly used by JavaScript framework
+  if ("NODE_ENV" in process.env) {
+    isDisabled |= process.env["NODE_ENV"] === "production" || process.env["NODE_ENV"] === "test";
+  }
+
+  // If the user has specifically opted-out or if we're running in a CI environment
+  isDisabled |= "REALM_DISABLE_ANALYTICS" in process.env || "CI" in process.env;
+
+  return isDisabled;
+}
+
+/**
+ * Send collected analytics data to Realm's servers over HTTPS
+ * @param  {Object} payload analytics info
+ */
+async function dispatchAnalytics(payload) {
+  const https = require("https");
+
+  return new Promise((resolve, reject) => {
+    const requestUrl = getAnalyticsRequestUrl(payload);
+
+    https
+      .get(requestUrl, (res) => {
+        resolve({
+          statusCode: res.statusCode,
+          statusMessage: res.statusMessage,
+        });
+      })
+      .on("error", (error) => {
+        const message = error && error.message ? error.message : error;
+        const err = new Error(`Failed to dispatch analytics: ${message}`);
+        reject(err);
+      });
+  });
+}
+
+async function submitAnalytics(dryRun) {
+  const wd = process.cwd();
+  const index = wd.indexOf("node_modules");
+  const packageJson = (index === -1 ? wd : wd.slice(0, index)) + "/package.json";
+  const context = require(packageJson);
+  const payload = await fetchPlatformData(context);
+  doLog(`payload: ${JSON.stringify(payload)}`);
+
+  if (dryRun) {
+    doLog("Dry run; will not submit analytics");
+    return;
+  }
+
+  if (isAnalyticsDisabled()) {
+    doLog("Analytics is disabled");
+    return;
+  }
+
+  await Promise.all([dispatchAnalytics(payload)]);
+}
+
+const optionDefinitions = [
+  { name: "dryRun", type: Boolean, multiple: false, description: "If true, don't submit analytics" },
+  { name: "log", type: Boolean, multiple: false, description: "If true, print log messages" },
+  { name: "test", type: Boolean, multiple: false, description: "If true, run as --dryRun --log" },
+];
+
+const options = commandLineArgs(optionDefinitions, { camelCase: true });
+
+let dryRun = false;
+if (options.dryRun) {
+  dryRun = true;
+}
+
+let log = false;
+if (options.log) {
+  log = true;
+}
+
+if (options.test) {
+  dryRun = true;
+  log = true;
+}
+
+if (log) {
+  doLog = (msg) => console.log(msg);
+} else {
+  // eslint-disable-next-line no-unused-vars
+  doLog = () => {
+    /* don't log */
+  };
+}
+
+(async function () {
+  await submitAnalytics(dryRun);
+})();
```

Comment:
This might result on an unhandled promise rejection.

```suggestion
submitAnalytics(dryRun).catch(console.error);
```

Reply:
Yeah, we don't want to draw attention if it doesn't work - for example if our end-point is down.